### PR TITLE
doc: reword iter module example and mention other methods

### PR DESCRIPTION
**Pull request by tesuji**

### CODE REVIEW

File: src/libcore/iter/mod.rs [hunk 1]
```diff
@@ -253,20 +243,23 @@
...
 //!     println!("{}", x);
 //! }
 //! ```
 //!
 //! [`map`]: trait.Iterator.html#method.map
+//! [`for_each`]: trait.Iterator.html#method.for_each
 //!
-//! The two most common ways to evaluate an iterator are to use a `for` loop
-//! like this, or using the [`collect`] method to produce a new collection.
+//! Another common way to evaluate an iterator are to use the [`collect`]
```
Comment (Centril) [inline src/libcore/iter/mod.rs]:
```suggestion
//! Another common way to evaluate an iterator is to use the [`collect`]
```

File: src/libcore/iter/mod.rs [hunk 2]
```diff
@@ -118,26 +118,16 @@
...
-//!
-//! let x = counter.next().unwrap();
-//! println!("{}", x);
-//!
-//! let x = counter.next().unwrap();
-//! println!("{}", x);
-//!
-//! let x = counter.next().unwrap();
-//! println!("{}", x);
+//! assert_eq!(counter.nex(), Some(1));
```
Comment (Centril) [inline src/libcore/iter/mod.rs]:
```suggestion
//! assert_eq!(counter.next(), Some(1));
```

File: src/libcore/iter/mod.rs [hunk 3]
```diff
@@ -253,20 +243,23 @@
 //! ```
 //!
 //! The idiomatic way to write a [`map`] for its side effects is to use a
-//! `for` loop instead:
+//! `for` loop or call [`for_each`] method:
```
Comment (Centril) [inline src/libcore/iter/mod.rs]:
```suggestion
//! `for` loop or call the [`for_each`] method:
```

**Comment by Centril** [comment]
r=me with green CI and suggestion applied.

**Comment by lzutao** [comment]
Ping

**Pull request merged**